libr/anal/esil.c

```diff
@@ -1786,9 +1787,12 @@ static int esil_poke_n(RAnalEsil *esil, int bits) {
 					ret = r_anal_esil_mem_write (esil, addr, b, bytes);
 					r_write_ble (b, num2, esil->anal->big_endian, 64);
 					ret = r_anal_esil_mem_write (esil, addr + 8, b, bytes);
-					return ret;
+					free (src2);
+					goto out;
```

Comment:
what about just removing the double `free(src2) and goto out` and use that only one time at the end of the if? the ony difference is in the `ret` value.